data-pipeline/user-cache-updater/src/main/java/org/ekstep/ep/samza/service/UserCacheUpdaterService.java

```diff
@@ -0,0 +1,181 @@
+                updateUserCache(event, userId, sink);
+            }
+            else { sink.markSkipped(); }
+        } catch (JsonSyntaxException e) {
+            LOGGER.error(null, "INVALID EVENT: " + source.getMessage());
+            sink.error();
+        }
+    }
+
+    public Map<String, Object> updateUserCache(Event event, String userId, UserCacheUpdaterSink sink) {
+        Map<String, Object> userCacheData = new HashMap<>();
+        try {
+            String userState = event.getUserStateValue();
+            if("Update".equals(userState)) {
+                String data = userDataStoreConnection.get(userId);
+                if (data != null && !data.isEmpty()) {
+                    userCacheData = gson.fromJson(data, mapType);
+
+                }
+                userCacheData.putAll(getUserLogininType(event));
+                ArrayList<String> userUpdatedList = event.getUserMetdataUpdatedList();
+                if (userCacheData.containsKey("usersignintype") && !"Anonymous".equals(userCacheData.get("usersignintype"))
+                        && null != userUpdatedList && !userUpdatedList.isEmpty()) {
+
+                    List<Row> userDetails = fetchFallbackDetailsFromDB(userId, userCacheUpdaterConfig.cassandra_user_table());
+                    Map<String, Object> userMetadataInfoMap = getUserMetaDataInfo(userDetails);
+
+                    if (!userMetadataInfoMap.isEmpty())
+                        userCacheData.putAll(userMetadataInfoMap);
+
+                    if (userUpdatedList.contains("id") && null != userCacheData.get("locationids")) {
+                        List<String> locationIds = (List<String>) userCacheData.get("locationids");
+
+                        List<Row> locationDetails = fetchFallbackDetailsFromDB(locationIds, userCacheUpdaterConfig.cassandra_location_table());
+                        Map<String, Object> userLocationMap = getLocationInfo(locationDetails);
+
+                        if (!userLocationMap.isEmpty())
+                            userCacheData.putAll(userLocationMap);
+                    }
+                }
+            }
+            else if("Create".equals(userState)) {
+                userCacheData.putAll(getUserSigninType(event));
+            }
+
+            if (!userCacheData.isEmpty()) {
+                redisConnect.addJsonToCache(userId, gson.toJson(userCacheData), userDataStoreConnection);
+                sink.success();
+            }
+
+        } catch (JedisException ex) {
+            sink.error();
+            LOGGER.error("", "Exception when adding to user redis cache", ex);
+            redisConnect.resetConnection();
+            try (Jedis redisConn = redisConnect.getConnection(userStoreDb)) {
+                this.userDataStoreConnection = redisConn;
+                if (null != userCacheData)
+                    redisConnect.addJsonToCache(userId, gson.toJson(userCacheData), userDataStoreConnection);
+            }
+        }
```

Comment:
Move the retry logic to a base class. Refer to the related comment in ContentCacheUpdaterService